bundles/io/org.openhab.io.caldav/src/main/java/org/openhab/io/caldav/internal/CalDavLoaderImpl.java

```diff
@@ -140,96 +142,100 @@ public void shutdown() {
-                }
-                String value = config.get(key) + "";
-
-                calDavConfig.setKey(id);
-                if (paramKey.equals(PROP_USERNAME)) {
-                    calDavConfig.setUsername(value);
-                } else if (paramKey.equals(PROP_PASSWORD)) {
-                    calDavConfig.setPassword(value);
-                } else if (paramKey.equals(PROP_URL)) {
-                    calDavConfig.setUrl(value);
-                } else if (paramKey.equals(PROP_RELOAD_INTERVAL)) {
-                    calDavConfig.setReloadMinutes(Integer.parseInt(value));
-                } else if (paramKey.equals(PROP_PRELOAD_TIME)) {
-                    calDavConfig.setPreloadMinutes(Integer.parseInt(value));
-                } else if (paramKey.equals(PROP_HISTORIC_LOAD_TIME)) {
-                    calDavConfig.setHistoricLoadMinutes(Integer.parseInt(value));
-                } else if (paramKey.equals(PROP_LAST_MODIFIED_TIMESTAMP_VALID)) {
-                    calDavConfig.setLastModifiedFileTimeStampValid(BooleanUtils.toBoolean(value));
-                } else if (paramKey.equals(PROP_DISABLE_CERTIFICATE_VERIFICATION)) {
-                    calDavConfig.setDisableCertificateVerification(BooleanUtils.toBoolean(value));
-                } else if (paramKey.equals(PROP_CHARSET)) {
-                    try {
-                        Charset.forName(value);
-                        calDavConfig.setCharset(value);
-                    } catch (UnsupportedCharsetException e) {
-                        log.error("charset not valid: {}", value);
-                    }
-                }
+        Map<String, CalDavConfig> configMap = new HashMap<String, CalDavConfig>();
+
+        Enumeration<String> iter = config.keys();
+        while (iter.hasMoreElements()) {
+            String key = iter.nextElement();
+
+            if (key.equals("service.pid")) {
+                continue;
             }
 
-            // verify if all required parameters are set
-            for (String id : configMap.keySet()) {
-                if (configMap.get(id).getUrl() == null) {
-                    throw new ConfigurationException(PROP_URL, PROP_URL + " must be set");
-                }
-                if (configMap.get(id).getUsername() == null) {
-                    throw new ConfigurationException(PROP_USERNAME, PROP_USERNAME + " must be set");
+            log.trace("processing configuration parameter: {}", key);
+            if (key.equals(PROP_TIMEZONE)) {
+                String newTimeZone = Objects.toString(config.get(key), null);
+                log.debug("overriding default timezone {} with {}", defaultTimeZone, newTimeZone);
+                defaultTimeZone = DateTimeZone.forID(newTimeZone);
+                if (defaultTimeZone == null) {
+                    log.warn("Invalid timezone value: {}", newTimeZone);
+                    throw new ConfigurationException("CalDAV IO", "Invalid timezone value: " + newTimeZone);
```

Comment:
Since we have/use a default timezone already, should we really throw an exception if the timezone in the configuration file isn't valid? Wouldn't it be better to log a warning and proceed to use the default?